### PR TITLE
Update Core / InitializeWorkflow

### PR DESCRIPTION
**Pull request by Sushisource**
## What was changed
Update core & handle start workflow rename. This _just_ deals with the rename, it does not attempt to use the new job application scheme discussed in https://github.com/temporalio/sdk-python/issues/606

## Why?
Need Core updates

## Checklist

1. Closes 

2. How was this tested:
Existing tests

3. Any docs updates needed?

### CODE REVIEW

File: temporalio/worker/_workflow_instance.py
Comment (cretz) [inline temporalio/worker/_workflow_instance.py:357]: We changed the order this item appears in the activation job list, correct? Can you confirm that that will have no effect here because of the built in Python ordering already present here?

**Review by cretz — COMMENTED** [review]
LGTM, just one ordering confirmation. Would also like @dandavison to look.

File: temporalio/worker/_workflow_instance.py
Comment (Sushisource) [reply temporalio/worker/_workflow_instance.py:357]: Yeah, it doesn't matter because everything gets reordered anyway

**Review by cretz — APPROVED** [review]
Please wait for @dandavison approval as well on this one, want him to make sure he doesn't see any issues.

**Review by dandavison — APPROVED** [review]
LGTM too. I can't see how https://github.com/temporalio/sdk-core/commit/54625225302d11463f11c0cdb7db6d2fca53e123 would change Python behavior given the lang-level control over job ordering.

**Pull request merged**
